src/time/Waveform.hpp

```diff
@@ -0,0 +1,122 @@
+#pragma once
+
+#include <Eigen/Core>
+#include "logging/Logger.hpp"
+
+namespace precice {
+
+namespace testing {
+// Forward declaration to friend the boost test struct
+class WaveformFixture;
+} // namespace testing
+
+namespace time {
+
+/**
+ * @brief Stores data samples in time and allows to perform interpolation to obtain new values.
+ */
```

Comment:
For an A+ documentation, you could add a small code example here.

Reply:
I would rather not add more documentation here. Especially untested examples are dangerous, because they easily get outdated. There are tests which serve as an example and they are per definition tested and work.